src/__tests__/codegenTypedDocuments.test.ts

```diff
@@ -138,4 +138,119 @@ describe("codegenTypedDocuments", () => {
       }"
     `);
   });
+
+  describe("module path customization", () => {
+    const queryDocument = parse(`
+      query authors {
+        authors {
+          idField
+        }
+      }
+    `);
+
+    const mutationDocument = parse(`
+      mutation createAuthor {
+        createAuthor {
+          idField
+        }
+      }
+    `);
+
+    const documents = [
+      { document: queryDocument, location: "literary/types/authors.gql" },
+      { document: mutationDocument, location: "mutations/createAuthor.gql" },
+    ];
+
+    const getPathConfig = (pluginConfig: Record<string, unknown>) => {
+      return getConfig({
+        documents,
+        plugins: [
+          {
+            codegenTypedDocuments: {
+              typesModule: "@codegen-types",
+              ...pluginConfig,
+            },
+          },
+        ],
+      });
+    };
+
+    it("wildcards the basename by default", async () => {
+      const config = getPathConfig({});
+      const output = await codegen(config);
+
+      expect(output).toMatchInlineSnapshot(`
+        "declare module \\"*/authors.gql\\" {
+          import { TypedDocumentNode } from \\"apollo-typed-documents\\";
+          import { AuthorsQuery, AuthorsQueryVariables } from \\"@codegen-types\\";
+          export const authors: TypedDocumentNode<AuthorsQueryVariables, AuthorsQuery>;
+          export default authors;
+        }
+
+        declare module \\"*/createAuthor.gql\\" {
+          import { TypedDocumentNode } from \\"apollo-typed-documents\\";
+          import { CreateAuthorMutation, CreateAuthorMutationVariables } from \\"@codegen-types\\";
+          export const createAuthor: TypedDocumentNode<CreateAuthorMutationVariables, CreateAuthorMutation>;
+          export default createAuthor;
+        }"
+      `);
+    });
+
+    it("respects the relativeToCwd setting", async () => {
+      const config = getPathConfig({ relativeToCwd: true });
+      const output = await codegen(config);
+
+      expect(output).toEqual(
+        expect.stringContaining(`declare module "*/literary/types/authors.gql"`)
+      );
+      expect(output).toEqual(
+        expect.stringContaining(`declare module "*/mutations/createAuthor.gql"`)
+      );
+    });
+
+    it("respects the prefix setting", async () => {
+      const config = getPathConfig({ prefix: "gql/" });
+      const output = await codegen(config);
+
+      expect(output).toEqual(
+        expect.stringContaining(`declare module "gql/authors.gql"`)
+      );
+      expect(output).toEqual(
+        expect.stringContaining(`declare module "gql/createAuthor.gql"`)
+      );
+    });
+
+    it("even respects the superfluous modulePathPrefix setting", async () => {
```

Comment:
Is it superfluous? I have never used it, the official docs says `Allows specifying a module definition path prefix to provide distinction between generated types.`, which sounds as it could be useful sometimes